### PR TITLE
Migrate to androix

### PR DESCRIPTION
**Pull request by martinbonnin**
This PR migrates to AndroidX and Android Studio 3.2.1.

It also adds the [gradle-versions-plugin](https://github.com/ben-manes/gradle-versions-plugin) and bumps several libraries to their latest version

### CODE REVIEW

File: cache/build.gradle [hunk 1]
```diff
@@ -60,6 +59,7 @@ dependencies {
     testImplementation cacheTestDependencies.junit
     testImplementation cacheTestDependencies.kotlinJUnit
     testImplementation cacheTestDependencies.mockito
+    testImplementation cacheTestDependencies.kotlinReflect
```
Comment (ghost) [inline cache/build.gradle:62]: I do not think that this library is necessary. Just curious - why did you add it in every module?

File: dependencies.gradle [hunk 2]
```diff
@@ -32,7 +31,7 @@ ext {
     espressoVersion = '3.0.0'
     testingSupportLibVersion = '0.1'
     mockitoKotlinVersion = '1.5.0'
-    mockitoAndroidVersion = '2.8.47'
+    mockitoAndroidVersion = '2.8.9' // As of today, 2.23.0 does not work
```
Comment (ghost) [inline dependencies.gradle:34]: downgraded library?

File: dependencies.gradle [hunk 3]
```diff
@@ -158,15 +162,13 @@ ext {
...
-            androidAnnotations: "com.android.support:support-annotations:${supportLibraryVersion}",
-            androidSupportV4:   "com.android.support:support-v4:${supportLibraryVersion}",
-            androidSupportV13:  "com.android.support:support-v13:${supportLibraryVersion}",
-            appCompatV7:        "com.android.support:appcompat-v7:${supportLibraryVersion}",
-            supportRecyclerView:"com.android.support:recyclerview-v7:${supportLibraryVersion}",
-            supportDesign:      "com.android.support:design:${supportLibraryVersion}",
+            androidAnnotations: "androidx.annotation:annotation:1.0.0",
+            appCompat:          "androidx.appcompat:appcompat:${appCompatVersion}",
+            supportRecyclerView:"androidx.recyclerview:recyclerview:1.0.0",
+            supportDesign:      "com.google.android.material:material:1.0.0-rc01",
```
Comment (ghost) [inline dependencies.gradle:171]: I am pretty sure that when you use `supportDesign` library you don't need to add `supportRecyclerView`.

File: mobile-ui/build.gradle [hunk 4]
```diff
@@ -1,3 +1,7 @@
+plugins {
```
Comment (ghost) [inline mobile-ui/build.gradle:1]: What is purpose of this plugin? Is it related to `Migrate to androidX` MR?

**Review by ghost — COMMENTED** [review]
I am not owner of this repository, but as a developer I wouldn't add any new libraries to any merge request without a note why it is important to this request or repository :-)

File: cache/build.gradle (hunk 1 above)
Comment (martinbonnin) [reply cache/build.gradle:62]: This is so long ago, I'm not sure any more... Maybe it has to do with a gradle warning saying that different versions of a lib were found on the classpath. Maybe it's something else. Feel free to remove it. If it works well then I guess it's not needed !

File: mobile-ui/build.gradle (hunk 4 above)
Comment (martinbonnin) [reply mobile-ui/build.gradle:1]: It's used to check if new versions of libraries are available. No needed for androidx. You can skip it.

**Comment by martinbonnin** [comment]
Closing due to inactivity
